internal/ns/nslogic.go

```diff
@@ -259,287 +243,175 @@ func copySecret(clientset *kubernetes.Clientset, secretName, operatorNamespace,
 	return nil
 }
 
-// ReconcileTargetRBAC ensures the RBAC resources in a target namespace are valid, reconciling
-// (i.e. updating or recreating) them if not.
-func ReconcileTargetRBAC(clientset *kubernetes.Clientset, pgoNamespace,
-	targetNamespace string) error {
+// ReconcileRole reconciles a Role required by the operator in a target namespace
+func ReconcileRole(clientset kubernetes.Interface, role, targetNamespace string,
+	roleTemplate *template.Template) error {
 
-	var errs []string
+	var createRole bool
 
-	// Use the image pull secrets of the operator service account in the new namespace.
-	operator, err := clientset.CoreV1().ServiceAccounts(pgoNamespace).Get(
-		OPERATOR_SERVICE_ACCOUNT, metav1.GetOptions{})
+	currRole, err := clientset.RbacV1().Roles(targetNamespace).Get(
+		role, metav1.GetOptions{})
 	if err != nil {
-		log.Errorf("expected the operator account to exist: %v", err)
-		errs = append(errs, err.Error())
+		if kerrors.IsNotFound(err) {
+			log.Debugf("Role %s in namespace %s does not exist and will be created",
+				role, targetNamespace)
+			createRole = true
```

Comment:
Is it possible to treat both create and update the same in these functions? My understanding is that clientset update becomes REST PUT, which I would expect to mean create-or-replace.

If that is so, then the not-found case can fallthrough to the diff which then triggers the update call, reducing some of the branching/complexity here. 🤞

Reply:
Like the idea here, and would like to try it out.  However, per our discussion will defer this to a future commit.